drf_extra_fields/fields.py

```diff
@@ -234,40 +242,56 @@ def get_initial(self):
         return self.to_representation(initial)
 
 
-class IntegerRangeField(RangeField):
+class PostgresDependencyCheckerMixin:
+    """ Mixin to check if postgres is none.
+    """
+    def check(self, **kwargs):
+        errors = super().check(**kwargs)
+        if postgres_fields is None:
+            errors.append(
+                checks.Error(
+                    "'psgl2' is required to use {name}".format(name=self.__class__.__name__),
+                    hint="Install the 'psycopg2' library from 'pip'",
+                    obj=self
+                )
+            )
+        return errors
+
+
+class IntegerRangeField(RangeField, PostgresDependencyCheckerMixin):
     child = IntegerField()
     range_type = NumericRange
 
 
-class FloatRangeField(RangeField):
+class FloatRangeField(RangeField, PostgresDependencyCheckerMixin):
```

Comment:
I like the idea but i think this will not work. These are not model fields, they are DRF fields. So i assume `check` method will not run. Am i missing something?

Reply:
oh :/ sorry my bad, the previous commit works. I'll try something similar. thank you for feedback !